18F/identity-idp
# LG-14655: A/B test to recommend platform authenticator to SMS users

## 🎫 Ticket

[LG-14655](https://cm-jira.usa.gov/browse/LG-14655)

## 🛠 Summary of changes

Implements new A/B test to recommend Face or Touch Unlock as an authenticator option after either...

- The user chooses only SMS as their MFA methods when creating an account
- The user signs in and MFAs with SMS and doesn't already have Face or Touch Unlock associated with their account

## 📜 Testing Plan

Force A/B test by adding configuration to `config/application.yml`:

```
recommend_webauthn_platform_for_sms_ab_test_percent: 100
```

Verify you are presented recommendation when creating account with only SMS:

1. On a device eligible to set up Face or Touch Unlock (iPhone, Android), go to http://localhost:3000
   - To simplify, you can use [Chrome DevTools device mode](https://developer.chrome.com/docs/devtools/device-mode) to simulate iPhone
2. Click "Create an account"
3. Continue account creation until MFA selection screen
4. Choose "Text or voice message" as your only selected MFA
5. Continue setting up phone
6. Observe that you arrive at the "Set up face or touch unlock for a more secure sign in" screen
7. Click either "Set up face or touch unlock" or "Skip"
8. Observe that you arrive where you expect
   - "Set up face or touch unlock": Face or touch unlock setup screen
   - "Skip": After-sign-in path, account page in this scenario

Verify you are presented recommendation when creating account with only SMS:

1. On `main` branch, repeat Steps 1-5 above
2. Click "Skip for now" when offered to set up second MFA
3. Click "Forget all browsers" on account dashboard and confirm prompt
4. Click "Sign out"
5. Switch back to `aduth-lg-14655-sms-ft-recommend-ab` branch
6. Sign in with the account you created
7. Authenticate with SMS
8. Repeat Steps 6-8 above

## 👀 Screenshots

<img src="https://github.com/user-attachments/assets/266096cb-2719-49ab-b4a3-b657add1ab23" alt="screenshot of recommendation screen" width="300">
